common/models/user.js

```diff
@@ -668,6 +668,45 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    var oldEmail, newEmail, oldId, newId;
+    UserModel.observe('before save', function(ctx, next) {
+      if (ctx.instance) {
+        if (ctx.instance.id !== undefined) {
+          ctx.hookState.id = ctx.instance.id;
+          oldId = ctx.hookState.id;
+        }
+      }
+      if (ctx.data) {
+        ctx.hookState.id = ctx.where.id;
+        oldId = ctx.hookState.id;
+      }
+      ctx.Model.find({ where: { id: oldId }}, function(err, userInstances) {
+        userInstances.forEach(function(userInstance) {
+          ctx.hookState.email = userInstance.email;
+          oldEmail = ctx.hookState.email;
+        });
+      });
+      next();
+    });
+
+    UserModel.observe('after save', function(ctx, next) {
+      var AccessToken = ctx.Model.relations.accessTokens.modelTo;
+      newEmail = ctx.instance.email;
+      newId = ctx.instance.id;
+      if (!ctx.isNewInstance && oldEmail !== newEmail && oldId === newId) {
+        AccessToken.deleteAll({ userId: ctx.instance.id }, function(err, info) {
```

Comment:
As I commented earlier, there may be more than a single user affected by a "save" operation. You should invalidate access tokens for all affected users, preferably via a single query.

```js
var newEmail = (ctx.instance || ctx.data).email;
var idsToExpire = ctx.hookState.originalUserData.filter(function(u) {
  return u.email !== newEmail;
}).map(fuction(u) {
  return u.id;
});
if (!idsToExpire.length)
  return next()l

AccessToken.deleteAll({ userId: { inq: idsToExpire }}, function(err, info) {
  // ...
});
```